quizgen/latex.py

```diff
@@ -3,32 +3,81 @@
 import shutil
 import subprocess
 
+import quizgen.util.dirent
+
 _pdflatex_bin_path = None
+_pdflatex_use_docker = False
+
+DOCKER_IMAGE = "quizgen/latex.py"
 
 def set_pdflatex_bin_path(path):
     global _pdflatex_bin_path
     _pdflatex_bin_path = path
 
-def is_available():
-    if (_pdflatex_bin_path is not None):
-        return True
-
-    if (shutil.which('pdflatex') is None):
-        logging.warning("Could not find `pdxlatex`, cannot compile PDFs")
-        return False
+def set_pdflatex_use_docker(use_docker):
+    global _pdflatex_use_docker
+    _pdflatex_use_docker = use_docker
 
-    return True
+def is_available(use_docker = False):
+    if use_docker:
+        result = subprocess.call(["docker", "--version"], stdout = subprocess.DEVNULL, stderr = subprocess.DEVNULL)
+        return result == 0
+    return _pdflatex_bin_path is not None or shutil.which('pdflatex') is not None
 
 def compile(path):
-    bin_path = 'pdflatex'
-    if (_pdflatex_bin_path is not None):
-        bin_path = _pdflatex_bin_path
+    if _pdflatex_use_docker:
+        return _compile_docker(path)
+    return _compile_local(path)
+
+def _compile_local(path):
+    bin_path = _pdflatex_bin_path or "pdflatex"
```

Comment:
In this project, we don't use `or` in a non-logical context.
(We also don't like to use implicit boolean conversion.)

Leave this how it was before.